services/components.py

```diff
@@ -72,3 +75,41 @@ def head_totals(self) -> ReportTotals:
     def patch_totals(self) -> ReportTotals:
         git_comparison = self.comparison.git_comparison
         return self.head_report.apply_diff(git_comparison["diff"])
+
+
+class ComponentMeasurements:
+    def __init__(
+        self,
+        raw_measurements: Iterable[dict],
+        component_id: str,
+        interval: Interval,
+        after: datetime,
+        before: datetime,
+    ):
+        self.raw_measurements = raw_measurements
+        self.component_id = component_id
+        self.interval = interval
+        self.after = after
+        self.before = before
+
+    @cached_property
+    def name(self):
+        return self.component_id
+
+    @cached_property
+    def percent_covered(self):
+        if len(self.raw_measurements) > 1:
```

Comment:
do we need to worry about return values for percent_covered or percent_changed if raw_measurements is 1 or 0?

Reply:
Yes, good catch, it should be ` if raw_measurements > 0`